src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/model/maven/Pom.java

```diff
@@ -48,6 +48,10 @@ public Pom(InputStream pomFileStream) throws IOException {
             this.gav = createGav(xPath, xmlDocument, "/project");
             this.parent = createGav(xPath, xmlDocument, "/project/parent");
 
+            if (!gav.isValid()) {
```

Comment:
>isValid [](http://example.com/codeflow?start=21&length=7)

Should we also add Scope and Type? Since they are used in case the BOM is referring another BOM?

Reply:
In this sense, I'm just using 'isValid' to represent whether we have successfully parsed the Maven GAV out of the XML file, so that we know whether we are looking at a Maven POM or just a random XML file.

Reply:
Ahh, where does the actual diffing happens? Is that done automatically once tokens are loaded via analyzer? I will look through the project to get the context there :)

Reply:
Yes, diffing is not done in 'my' code, all 'my' code does is tokenise the given input (xml file or java source files) into json that are then fed into APIView. The diffing is performed by APIView itself after it receives the JSON file.